projects/RabbitMQ.Client/client/impl/ConsumerDispatching/ConsumerDispatcherBase.cs

```diff
@@ -8,7 +10,7 @@ namespace RabbitMQ.Client.ConsumerDispatching
     internal abstract class ConsumerDispatcherBase
     {
         private static readonly FallbackConsumer s_fallbackConsumer = new FallbackConsumer();
-        private readonly Dictionary<string, IBasicConsumer> _consumers = new Dictionary<string, IBasicConsumer>();
+        private readonly IDictionary<string, IBasicConsumer> _consumers = new ConcurrentDictionary<string, IBasicConsumer>();
```

Comment:
Do not erase type information on private members as it hinders the capacity of the compiler to generate performant code. It can also lead to expensive code being written, like invoking `Count` and enumerating the dictionary.

```suggestion
        private readonly ConcurrentDictionary<string, IBasicConsumer> _consumers = new ConcurrentDictionary<string, IBasicConsumer>();
```

Reply:
Thank you!